test/elm/interval/test.coffee

```diff
@@ -1405,3 +1405,699 @@ describe 'Collapse', ->
+    e = "{ [T01:00:00.000+00:00, T01:00:00.999+00:00], [T01:00:01.000+00:00, T01:00:01.999+00:00] }"
+    prettyList(@msPrecPerSecond.exec(@ctx)).should.equal e
+    
+    # define MsPrecPerMillisecond: expand { Interval[@T01:00:00.000+00:00, @T01:00:00.002+00:00] } per millisecond
+    e = "{ [T01:00:00.000+00:00, T01:00:00.000+00:00], [T01:00:00.001+00:00, T01:00:00.001+00:00] }"
+    prettyList(@msPrecPerMillisecond.exec(@ctx)).should.equal e
+
+  xit 'expands a second precision datetime', ->
+    # define SecPrecPerHour: expand { Interval[@T01:00:00+00:00, @T03:00:00+00:00] } per hour
+    e = "{ [T01:00:00+00:00, T01:59:59+00:00], [T02:00:00+00:00, T02:59:59+00:00] }"
+    prettyList(@secPrecPerHour.exec(@ctx)).should.equal e
+    
+    # define SecPrecPerMinute: expand { Interval[@T01:00:00+00:00, @T01:02:00+00:00] } per minute
+    e = "{ [T01:00:00+00:00, T01:00:59+00:00], [T01:01:00+00:00, T01:01:59+00:00] }"
+    prettyList(@secPrecPerMinute.exec(@ctx)).should.equal e
+    
+    # define SecPrecPerSecond: expand { Interval[@T01:00:00+00:00, @T01:00:01+00:00] } per second
+    e = "{ [T01:00:00+00:00, T01:00:00+00:00], [T01:00:01+00:00, T01:00:01+00:00] }"
+    prettyList(@secPrecPerSecond.exec(@ctx)).should.equal e
+
+    should.not.exist @secPrecPerMillisecond.exec(@ctx)
+  
+  xit 'expands a minute precision datetime', ->
+    # define MinPrecPerHour: expand { Interval[@T01:00+00:00, @T03:00+00:00] } per hour
+    e = "{ [T01:00+00:00, T01:59+00:00], [T02:00+00:00, T02:59+00:00] }"
+    prettyList(@minPrecPerHour.exec(@ctx)).should.equal e
+    
+    # define MinPrecPerMinute: expand { Interval[@T01:00+00:00, @T01:02+00:00] } per minute
+    e = "{ [T01:00+00:00, T01:00+00:00], [T01:01+00:00, T01:01+00:00] }"
+    prettyList(@minPrecPerMinute.exec(@ctx)).should.equal e
+    
+    should.not.exist @minPrecPerSecond.exec(@ctx)
+    should.not.exist @minPrecPerMillisecond.exec(@ctx)
+  
+  xit 'expands an hour precision datetime', ->
+    # define HourPrecPerHour: expand { Interval[@T01+00:00, @T02+00:00] } per hour
+    e = "{ [T01+00:00, T01+00:00], [T02+00:00, T02+00:00] }"
+    prettyList(@hourPrecPerHour.exec(@ctx)).should.equal e
+    
+    should.not.exist @hourPrecPerMinute.exec(@ctx)
+    should.not.exist @hourPrecPerSecond.exec(@ctx)
+    should.not.exist @hourPrecPerMillisecond.exec(@ctx)
+
+describe 'QuantityIntervalExpand', ->
+  @beforeEach ->
+    setup @, data
+
+  it 'expands single intervals', ->
+    # define ClosedSingleGPerG: expand { Interval[2 'g', 4 'g'] } per 1 'g'
+    a = @closedSingleGPerG.exec(@ctx)
+    prettyList(a).should.equal "{ [2 'g', 2 'g'], [3 'g', 3 'g'], [4 'g', 4 'g'] }"
+    
+    # define ClosedSingleGPerGDecimal: expand { Interval[2.1 'g', 4.1 'g'] } per 1 'g'
+    a = @closedSingleGPerGDecimal.exec(@ctx)
+    prettyList(a).should.equal "{ [2.1 'g', 3.09999999 'g'], [3.1 'g', 4.09999999 'g'] }"
```

Comment:
OK -- although it would be cleaner to use an open interval in the result.  E.g.
```
{ [2.1 'g', 3.1 'g'), [3.1 'g', 4.1 'g') }
```
That said, according to spec and the defined precision (8 decimals), what you have is technically equivalent.  So don't feel you _have_ to change this if it presents difficult challenges.